LandSandBoat/server
# Add magic enfeebling resist building (that decreases effect duration)

**_I affirm:_**

## What does this pull request do?
This PR adds an enfeebling resistance building system to the new magic framework. The resistance building is based on retail captures of Fafnir by Jimmayus and Kanryu (from 2019 and 2023). 

The building system decreases the final duration of the status effect by X seconds for each successful cast of a spell that adds that effect to the mob. The value X is set via a mod, so for fafnir for SLEEP this would be `mob:setMod(xi.mod.SLEEPRESBUILD, 3)` , as the retail captures show a decrease of 3 seconds. The system adds NAMERESBUILD mods for many enfeebles to allow res building to different status effects, and leverages local variables to track the amount of resistance building.

This built up resistance potentially should be cleared if the mob starts roaming or over some amount of time, but this is not verified so is not yet implemented. This system is also useful for custom private servers that want to prevent, for example, players from holding mobs asleep indefinitely.

Related to this bug report https://github.com/LandSandBoat/server/issues/847

This is coming from ASB to upstream.

## Steps to test these changes
Pop Fafnir, cast sleep, wait for it to wear off, the cast again and see the length decrease in duration by 3 seconds, repeat.
